### ISSUE TITLE
Export ExpiringResource from /sdk/internal so SDKs can implement their own authentication policy

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-sdk-for-go#17752]
### Feature Request

The bearer token policy uses a hard-coded format string for populating the `Authentication` header value.  This means that the policy can't be used for other services that use an OAuth token for authentication with a slightly different value (e.g. Cosmos).

We can add an option to the `BearerTokenPolicyOptions` type (which is current empty) that allows specifying an optional format string.

```go
// BearerTokenOptions configures the bearer token policy's behavior.
type BearerTokenOptions struct {
	// Format is the format string used when constructing the value for the Authentication header.
	// The empty string defaults to format "Bearer %s", where %s is replaced with the OAuth token.
	Format string
}
```

**Comment by ealsur** [comment]
For reference, in Cosmos we have a different Auth header format for AAD: https://github.com/Azure/azure-sdk-for-go/blob/4b68333d02a427eb47985a8c61b828fdc16b9dd4/sdk/data/azcosmos/cosmos_policy_bearer_token.go#L26

In the current state, the only alternative is to have a policy read the current Bearer Auth header and replace it with our format.

**Comment by chlowell** [comment]
Given that this wouldn't be generally useful, other languages haven't added a similar option, and that we have only one use case for it here, I think we should wait to see more need before adding it. Today's workarounds don't seem too bad to me. Here's another one:
```go
type cosmosCredentialWrapper struct {
	cred azcore.TokenCredential
}

func (c *cosmosCredentialWrapper) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (*azcore.AccessToken, error) {
	tk, err := c.cred.GetToken(ctx, opts)
	if err != nil {
		return nil, err
	}
	// this is okay because our auth policies handle tokens as opaque strings
	tk.Token = "type=aad&ver=1.0&sig=" + tk.Token
	return tk, err
}
```

**Comment by jhendrixMSFT** [comment]
This will preserve the `Bearer ` prefix though which is problematic for Cosmos.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/policy_bearer_token.go#L61

**Comment by jhendrixMSFT** [comment]
Good point about the RFC.  I think then the best option is to expose the necessary primitives like `ExpiringResource` so that SDK authors can implement their own policy.

**Comment by jhendrixMSFT** [comment]
Fixed in upcoming `internal@v0.10.0` release.
